File: finanz-api/src/main/java/com/finanzapi/model/Transaccion.java
package com.finanzapi.model;

import jakarta.persistence.*;
import java.time.LocalDate;

/**
 * JPA entity representing a single financial transaction.
 *
 * <p>Persisted in the H2 {@code transacciones} table. The schema is managed
 * by Hibernate via the active Spring profile ({@code ddl-auto=update} for dev,
 * {@code create-drop} for the test profile).</p>
 *
 * <p>The {@code fecha} field is always overwritten to today by
 * {@link com.finanzapi.service.BudgetService#registrar} — values supplied by
 * the client are ignored.</p>
 *
 * <p>This entity is the Spring Data / JPA counterpart of {@code finanz-core}'s
 * plain {@code Transaccion} class; both share the same field names and semantics.</p>
 */
@Entity
@Table(name = "transacciones")
public class Transaccion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String descripcion;
    private double monto;

    @Enumerated(EnumType.STRING)
    private TipoTransaccion tipo;

    private LocalDate fecha;

    public Transaccion() {}

    public Transaccion(String descripcion, double monto, TipoTransaccion tipo, LocalDate fecha) {
        this.descripcion = descripcion;
        this.monto = monto;
        this.tipo = tipo;
        this.fecha = fecha;
    }

    public Long getId() { return id; }
    public String getDescripcion() { return descripcion; }
    public void setDescripcion(String descripcion) { this.descripcion = descripcion; }
    public double getMonto() { return monto; }
    public void setMonto(double monto) { this.monto = monto; }
    public TipoTransaccion getTipo() { return tipo; }
    public void setTipo(TipoTransaccion tipo) { this.tipo = tipo; }
    public LocalDate getFecha() { return fecha; }
    public void setFecha(LocalDate fecha) { this.fecha = fecha; }
}

File: finanz-app/app/src/main/java/com/finanzapp/data/model/Transaccion.kt
package com.finanzapp.data.model

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDate

/**
 * Room entity representing a single financial transaction.
 *
 * Persisted in the `transacciones` table. [LocalDate] and [TipoTransaccion] are
 * stored via [com.finanzapp.data.local.Converters]. [Categoria] is embedded with
 * the `cat_` column prefix so its fields live in the same row.
 *
 * The [categoria] is always populated with a default `"General"` / `0.0` value by
 * [com.finanzapp.domain.BudgetService]; the UI does not expose it to the user.
 */
@Entity(tableName = "transacciones")
data class Transaccion(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val descripcion: String,
    val monto: Double,
    val tipo: TipoTransaccion,
    val fecha: LocalDate,
    @Embedded(prefix = "cat_")
    val categoria: Categoria
)
